addons/account/tests/test_account_inalterable_hash.py

```diff
@@ -676,3 +662,143 @@ def test_inalterable_hash_verification_by_batches(self):
 
         with patch('odoo.addons.account.models.company.INTEGRITY_HASH_BATCH_SIZE', 12):
             self._verify_integrity(moves, "Entries are correctly hashed", moves[0], moves[-1], moves[0].journal_id.name)
+
+    def test_error_on_unreconciled_bank_statement_lines(self):
+        """
+        Check that an error is raised when we try to hash entries with unreconciled bank statement lines.
+        """
+        unreconciled_bank_statement_line = self.env['account.bank.statement.line'].create({
+            'journal_id': self.company_data['default_journal_bank'].id,
+            'date': '2017-01-01',
+            'payment_ref': '2017_unreconciled',
+            'amount': 10.0,
+        })
+        unreconciled_move = unreconciled_bank_statement_line.move_id
+        unreconciled_move.journal_id.restrict_mode_hash_table = True
+        with self.assertRaisesRegex(UserError, "An error occurred when computing the inalterability. All entries have to be reconciled."), self.env.cr.savepoint():
+            unreconciled_move.button_hash()
+
+    def test_wizard_hashes_all_journals(self):
+        """
+        Test that the wizard hashes all journals.
+          * Regardless of the `restrict_mode_hash_table` setting on the journal.
+          * Regardless of the lock date
+        """
+        moves = self.env['account.move'].create([
+            {
+                'date': '2023-01-02',
+                'journal_id': self.env['account.journal'].create({
+                    'code': f'wiz{idx}',
+                    'name': f'Wizard {journal_type}',
+                    'type': journal_type,
+                }).id,
+                'line_ids': [Command.create({
+                    'name': 'test',
+                    'quantity': 1,
+                    'price_unit': 1000,
```

Comment:
A price unit is a bit weird on a journal entry, and we shouldn't rely on the automatic balance for tests (if it was even using it)
```suggestion
                    'balance': 0,
```